tests/dspa_v2_test.go

```diff
@@ -91,3 +91,52 @@ func (suite *IntegrationTestSuite) TestDSPADeployment() {
 		}
 	})
 }
+
+func (suite *IntegrationTestSuite) TestDSPADeploymentWithK8sNativeApi() {
+	suite.T().Run("should verify API Server configuration based on PipelineStorage", func(t *testing.T) {
+		timeout := time.Second * 120
+		interval := time.Second * 2
+
+		// First check if PipelineStorage is set to kubernetes
+		if suite.DSPA.Spec.APIServer.PipelineStorage != "kubernetes" {
+			t.Log("PipelineStorage is not set to kubernetes, skipping K8s Native API verification")
+			t.SkipNow()
+		}
+
+		// Verify API Server configuration
```

Comment:
Could you please also check that the webhook was deployed and running in this test as well>